common/domain/replicationTaskExecutor.go

```diff
@@ -210,9 +233,11 @@ func (h *domainReplicationTaskExecutorImpl) handleDomainUpdateReplicationTask(ct
 
 	// plus, we need to check whether the config version is <= the config version set in the input
 	// plus, we need to check whether the failover version is <= the failover version set in the input
-	existingDomain, err := h.domainManager.GetDomain(ctx, &persistence.GetDomainRequest{
```

Comment:
this gets mutated due to the substructs getting dereferenced and then edited, so it was necessary to deepcopy even though there's no actual editing of the existingDomain variable